### DISCUSSION TITLE
Welcome to esp_local_control Discussions!

### DISCUSSION THREAD

**Discussion by chmoore889** [chmoore889/esp_local_control#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by Fahadk7** [comment]
Hi
I have been trying to build a project with ESP32 to try out Local Control  but via BLE
i have done some beginner level work with ESP Rainmaker, so any help on BLE control would be appreaciated

Thanks!

**Reply by chmoore889** [reply]
Hi there. This repo and ESP Rainmaker as a whole only communicates with devices over WiFi. BLE can't be used at all by Rainmaker as far as I know. Controlling an ESP32 over BLE is pretty straightforward since there's less going on than in Rainmaker. You can take a look at the examples provided by ESP IDF [here](https://github.com/espressif/esp-idf/tree/master/examples/bluetooth/nimble/bleprph) which is how I learned how to first use BLE and there should be some tutorials for how to do this floating around elsewhere.

I would suggest first taking a look at all the documentation ESP IDF has to offer and give writing the code yourself a solid attempt. Then if you run into any problems, write up a question on Stack Overflow. I can't help out anymore on this discussion thread since your work isn't really related to this repo and I'm a bit short on time.

Best of luck!

**Reply by Fahadk7** [reply]
Hi Christopher, Thanks for the suggestion!
